command/operator/raft/listpeers/operator_raft_list.go

```diff
@@ -70,8 +70,29 @@ func raftListPeers(client *api.Client, stale bool) (string, error) {
 		return "", fmt.Errorf("Failed to retrieve raft configuration: %v", err)
 	}
 
+	autoPilotReply, err := client.Operator().GetAutoPilotHealth(q)
+	if err != nil {
+		return "", fmt.Errorf("Failed to retrieve autopilot health: %v", err)
+	}
+
+	serverHealthDataMap := make(map[string]api.ServerHealth)
+	leaderLastCommitIndex := uint64(0)
+
+	for _, serverHealthData := range autoPilotReply.Servers {
+		serverHealthDataMap[serverHealthData.ID] = serverHealthData
+	}
+
+	for _, s := range reply.Servers {
+		if s.Leader {
+			serverHealthDataLeader, ok := serverHealthDataMap[s.ID]
+			if ok {
+				leaderLastCommitIndex = serverHealthDataLeader.LastIndex
+			}
+		}
+	}
+
```

Comment:
Instead of making a second API request to get a single value (`LastIndex`) per server, how feasible would it be to just add `LastIndex` to the existing `RaftConfiguration` struct returned by the server side implementation of `RaftGetConfiguration(...)`. It seems the data is readily accessible via `consul.Server.autopilot.GetState().Servers[].Stats.LastIndex`. 

@dhiaayachi or @boxofrad Are there prior established patterns for this type of change that gets an additional piece of data or are multiple APIs calls a norm in the implemenation of `command`s?

Reply:
I agree with @analogue extending the `RaftConfiguration` to add the `LastIndex` seem to be a cleaner approach in this case. 

I don't think multiple API calls is a norm here and I would vote for having all the data needed included in a single API call as much as possible. It will also give the possibility to users to use the API to perform some automation, like watching the servers before going forward to the next step of a rolling upgrade.

Reply:
done.